XenAdmin/Wizards/NewPolicyWizard/NewPolicyWizardSpecific.cs

```diff
@@ -0,0 +1,303 @@
+            if (VMGroup<T>.isVMPolicyVMPP)
+            {
+                xenTabPageSnapshotFrequency = new NewPolicySnapshotFrequencyPage(false);
+                xenTabPageSnapshotFrequency.Pool = pool;
+                AddPages(xenTabPageSnapshotFrequency);
+
+                xenTabPageArchive = new NewPolicyArchivePage();
+                xenTabPageArchive.Pool = pool;
+                AddPage(xenTabPageArchive);
+
+                xenTabPageEmail = new NewPolicyEmailPage();
+                xenTabPageEmail.Pool = pool;
+                AddPages(xenTabPageEmail);
+
+            }
+            else /*VMSS*/
+            {
+                xenTabPageSnapshotFrequency = new NewPolicySnapshotFrequencyPage(true);
+                xenTabPageSnapshotFrequency.Pool = pool;
+                AddPages(xenTabPageSnapshotFrequency);
+            }
+            AddPages(xenTabPageFinish);
+        }
+
+        public NewPolicyWizardSpecific(Pool pool, List<VM> selection)
+            : this(pool)
+        {
+            this.xenTabPageVMsPage.SelectedVMs = selection;
+        }
+
+        private new string GetSummary()
+        {
+
+            if (VMGroup<T>.isVMPolicyVMPP)
+            {
+                return string.Format(Messages.POLICY_SUMMARY.Replace("\\n", "\n").Replace("\\r", "\r"), xenTabPagePolicy.PolicyName, CommaSeparated(xenTabPageVMsPage.SelectedVMs),
+                                     FormatBackupType(xenTabPageSnapshotType.BackupType),
+                                     FormatSchedule(xenTabPageSnapshotFrequency.Schedule, xenTabPageSnapshotFrequency.Frequency, DaysWeekCheckboxes.DaysMode.L10N_LONG),
+                                     FormatSchedule(xenTabPageArchive.Schedule, xenTabPageArchive.ArchiveFrequency, DaysWeekCheckboxes.DaysMode.L10N_LONG));
+            }
+            else
+            {
+                return string.Format(Messages.VMSS_POLICY_SUMMARY.Replace("\\n", "\n").Replace("\\r", "\r"), xenTabPagePolicy.PolicyName, CommaSeparated(xenTabPageVMsPage.SelectedVMs),
+                                     FormatBackupType(xenTabPageSnapshotType.BackupType),
+                                     FormatSchedule(xenTabPageSnapshotFrequency.Schedule, xenTabPageSnapshotFrequency.Frequency, DaysWeekCheckboxes.DaysMode.L10N_LONG));
+            }
+        }
+
+        private int GetBackupDaysCount()
+        {
+            string days;
+            return xenTabPageSnapshotFrequency.Schedule.TryGetValue("days", out days) ? days.Split(',').Length : 0;
+        }
+
+        private static string FormatBackupType(vmpp_backup_type backupType)
+        {
+            if (backupType == vmpp_backup_type.snapshot)
+                return Messages.DISKS_ONLY;
+            else if (backupType == vmpp_backup_type.checkpoint)
+                return Messages.DISKS_AND_MEMORY;
+            throw new ArgumentException("wrong argument");
+        }
+
+        private static string FormatBackupTypeVMSS(vmss_type backupType)
```

Comment:
This could be combined with the previous function by defining a new enum for backup type in IVMGroup.